content/300-guides/050-database/900-turso.mdx

```diff
@@ -0,0 +1,202 @@
+
+<Admonition type="warning">
+
+Support for Turso is available in [Early Access](/about/prisma/releases#early-access) from Prisma versions 5.4.2 and later.
+
+</Admonition>
+
+## Commonalities with other database providers
+
+libSQL is 100% compatible with SQLite. libSQL extends SQLite and adds the following features and capabilities:
+
+- Support for replication
+- Support for automated backups
+- Ability to embed Turso as part of other programs such as the Linux kernel
+- Supports user-defined functions
+- Support for asynchronous I/O
+
+> To learn more about the differences between libSQL and how it is different from SQLite, see [libSQL Manifesto](https://turso.tech/libsql-manifesto).
+
+Many aspects of using Prisma with Turso are just like using Prisma with any other relational database. You can still:
+
+- model your database with the [Prisma Schema Language](/concepts/components/prisma-schema)
+- use Prisma's existing [`sqlite` database connector](/concepts/database-connectors/sqlite) in your schema
+- use [Prisma Client](/concepts/components/prisma-client) in your application to talk to the database server at Turso
+
+## Differences to consider
+
+There are a number of differences between Turso and SQLite to consider. You should be aware of the following when deciding to use Turso and Prisma:
+
+- **Remote and embedded SQLite databases**. libSQL uses HTTP to connect to the remote SQLite database. libSQL also supports remote database replicas and embedded replicas. Embedded replicas enable you to replicate your primary database inside your application.
+- **Making schema changes**. Since libSQL uses HTTP to connect to the remote database, this makes it incompatible with Prisma Migrate. However, you can use [`prisma migrate diff`](/reference/api-reference/command-reference#migrate-diff) <span class="api"></span> to create a schema migration and then apply the changes to your database using [Turso's CLI](https://docs.turso.tech/reference/turso-cli).
+
+## Connect and query your primary database
+
+<br />
+
+<details>
+
+<summary>
+  Expand to learn how to provision a database and retrieve your credentials on
+  Turso
+</summary>
+
+Ensure that you have the [Turso CLI](https://docs.turso.tech/reference/turso-cli) installed to manage your databases.
+
+1. To provision a database on Turso, run the following command:
+
+```terminal
+turso db create turso-prisma-db
+```
+
+2. Retrieve the database's connection string:
+
+```terminal
+turso db show turso-prisma-db
+```
+
+3. Create an authentication token that will allow you to connect to the database:
+
+```terminal
+turso db tokens create turso-prisma-db
+```
+
+4. Update your `.env` file with the authentication token and connection string:
+
+```text file=.env
+TURSO_AUTH_TOKEN="eyJhbGciOiJFZERTQSIsInR5cCI6IkpXVCJ9..."
+TURSO_DATABASE_URL="libsql://turso-prisma-db-user.turso.io"
+```
```

Comment:
Indentation is broken.
<img width="137" alt="image" src="https://github.com/prisma/docs/assets/183673/104c6c0f-7b8e-4ace-8978-8b8c65e7ad67">